### ISSUE TITLE
[FEATURE] <Adding different way for authentication.>

### ISSUE THREAD

**Issue by AdityaPainuli** [ykdojo/defaang#85]
### Description

Can I start working on the authentication part of the sign-in and sign-up page and what are the platforms we will allow the user to authenticate themselves?

**Comment by rohitdasu** [comment]
Hi @AdityaPainuli thanks for raising this issue. We will definitely need social authentication as nowadays most users do authentication using social medias. In my opinion let's do with 'Google' and 'Github'. 
Would love your suggestion @ykdojo

**Comment by RakeshSangem** [comment]
I also believe that GitHub and Google are sufficient for authentication.

**Comment by AdityaPainuli** [comment]
I will start working on this feature soon.

**Comment by makyfj** [comment]
@AdityaPainuli are you planning to use https://next-auth.js.org/ for auth?

**Comment by AdityaPainuli** [comment]
No, I am using supebase authentication I think it will be good to use only the stack or packages which are really needed.

**Comment by nateemerson** [comment]
Next Auth could be helpful for integrating some of the auth flows w/ Next Middleware..but I'm not very familiar w/ Supabase auth. I'd be curious how the two connect / work together...I'm assuming there's still a way to leverage Supabase auth from Next middleware so probably not a dealbreaker one way or the other.

**Comment by ykdojo** [comment]
This one looks interesting, just came across it: https://themodern.dev/courses/build-a-fullstack-app-with-nextjs-supabase-and-prisma-322389284337222224

**Comment by ykdojo** [comment]
And this: https://twitter.com/supabase/status/1573100354174332928?s=46&t=ydHdma5FnBvGoSlDUOYE1g

**Comment by AdityaPainuli** [comment]
> This one looks interesting, just came across it: https://themodern.dev/courses/build-a-fullstack-app-with-nextjs-supabase-and-prisma-322389284337222224

Yes , I also come across that and currently I am learning how to make it work for one of my project
